gobblin-yarn/src/main/java/org/apache/gobblin/yarn/GobblinYarnAppLauncher.java

```diff
@@ -499,6 +504,14 @@ void stopYarnClient() {
   private Optional<ApplicationId> getApplicationId() throws YarnException, IOException {
```

Comment:
should this function name changed to `getOrCreateNewYarnApplication` ? just a thought